lib/turbolinks.rb

```diff
@@ -21,11 +21,13 @@ class Engine < ::Rails::Engine
           end
         end
 
-        ActionDispatch::Request.class_eval do
-          def referer
-            self.headers['X-XHR-Referer'] || super
+        unless Rails.version < '4.0'
```

Comment:
You shouldn't have to do this.  The reason that test failed is because `referer` in Rails 3 is defined in Rack and checks for the header named `HTTP_REFERER` not `Referer`.